lib/serviceClient.ts

```diff
@@ -143,6 +143,11 @@ export interface ServiceClientOptions {
    * HTTP and HTTPS agents which will be used for every HTTP request (Node.js only).
    */
   agentSettings?: AgentSettings;
+  /**
+   * If specified, this is the base URI that requests will be made against for this ServiceClient.
+   * If it is not specified, then all OperationSpecs must contain a baseUrl property.
```

Comment:
I see that we copied this from the docs for the corresponding protected property in ServiceClient. Now that we are adding one more use for it, that should be documented here and on that property in ServiceClient as well

Reply:
How does this look? https://github.com/Azure/ms-rest-js/pull/457/commits/557e53794a930b89c76eea396cb2a030724c5d73#diff-9f4919aa6f25f2ebd2d7e5f17c0d2c706cbc31691fc28e197ab9fd662a316763R147-R153